crypto/fipsmodule/ml_kem/ml_kem.c

```diff
@@ -34,16 +34,36 @@ int ml_kem_512_keypair_deterministic_no_self_test(uint8_t *public_key  /* OUT */
                                                   uint8_t *secret_key  /* OUT */,
                                                   const uint8_t *seed  /* IN */) {
   ml_kem_params params;
+  int res;
   ml_kem_512_params_init(&params);
-  return ml_kem_keypair_derand_ref(&params, public_key, secret_key, seed);
+  res = ml_kem_keypair_derand_ref(&params, public_key, secret_key, seed);
+#if defined(AWSLC_FIPS)
```

Comment:
Can this be implemented in `AWS_LC_FIPS_failure`?

Maybe can avoid littering the code-base with `if defined(AWS_FIPS)`.

Reply:
I stuck with the precedent here and prefer not to change it -- but it's ultimately your call.